sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -106,6 +106,8 @@ case class HashAggregateExec(
         // This is a grouped aggregate and the input iterator is empty,
         // so return an empty iterator.
         Iterator.empty
+      } else if (!hasInput && resultExpressions.isEmpty) {
```

Comment:
`val res = if (!hasInput && (groupingExpressions.nonEmpty || resultExpressions.isEmpty)) {`? in line 105? Also, we need to update the comment in line 106.